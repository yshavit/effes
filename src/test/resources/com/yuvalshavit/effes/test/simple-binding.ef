data type VoidWrap(value: Void)
data type BoolWrap(value: True | False)

def extractValue (wrapper: BoolWrap | VoidWrap) -> True | False | Void:
  return case wrapper of
    BoolWrap(tf): tf
    VoidWrap(v): v

def main -> Void:
  b1 = BoolWrap(True)
  b2 = BoolWrap(False)
  v1 = VoidWrap(Void)
  print extractValue b1
  print extractValue b2
  print extractValue v1
  return Void