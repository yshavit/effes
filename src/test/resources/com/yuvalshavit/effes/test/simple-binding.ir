def extractValue (wrapper: BoolWrap | VoidWrap) -> False | True | Void:
  ReturnStatement
    CaseExpression
      case:
        VarExpression
          arg: wrapper
          pos: 0
      of BoolWrap:
        VarExpression
          var: tf
          pos: 0
      of VoidWrap:
        VarExpression
          var: v
          pos: 0
