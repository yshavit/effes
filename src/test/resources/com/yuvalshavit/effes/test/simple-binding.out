Your argument was: True
Your argument was: False
Your argument was: Void
>> Void
